registry/application/services/service_publisher_service.py

```diff
@@ -0,0 +1,88 @@
+        if record_exist:
+            return ServiceAvailabilityStatus.UNAVAILABLE.value
+        return ServiceAvailabilityStatus.AVAILABLE.value
+
+    def save_service(self, payload):
+        service = ServiceFactory().create_service_entity_model(self.org_uuid, self.service_uuid, payload,
+                                                               ServiceStatus.DRAFT.value)
+        service = ServiceRepository().save_service(self.username, service, ServiceStatus.DRAFT.value)
+        return service.to_dict()
+
+    def create_service(self, payload):
+        service_uuid = uuid4().hex
+        service = ServiceFactory().create_service_entity_model(self.org_uuid, service_uuid, payload,
+                                                               ServiceStatus.DRAFT.value)
+        ServiceRepository().add_service(service, self.username)
+        return {"org_uuid": self.org_uuid, "service_uuid": service_uuid}
+
+    def get_services_for_organization(self, payload):
+        offset = payload.get("offset", DEFAULT_OFFSET)
+        limit = payload.get("limit", DEFAULT_LIMIT)
+        search_string = payload["q"]
+        search_attribute = payload["s"]
+        sort_by = payload["sort_by"].lower()
+        order_by = payload["order_by"].lower()
+        filter_parameters = {
+            "offset": offset,
+            "limit": limit,
+            "search_string": search_string,
+            "search_attribute": search_attribute if search_attribute in ALLOWED_ATTRIBUTES_FOR_SERVICE_SEARCH else DEFAULT_ATTRIBUTE_FOR_SERVICE_SEARCH,
+            "sort_by": sort_by if sort_by in ALLOWED_ATTRIBUTES_FOR_SERVICE_SORT_BY else DEFAULT_ATTRIBUTES_FOR_SERVICE_SORT_BY,
+            "order_by": order_by if order_by in ALLOWED_ATTRIBUTES_FOR_SERVICE_SORT_BY else DEFAULT_ATTRIBUTES_FOR_SERVICE_ORDER_BY
+        }
+        search_result = ServiceRepository().get_services_for_organization(self.org_uuid, filter_parameters)
+        search_count = ServiceRepository().get_total_count_of_services_for_organization(self.org_uuid,
+                                                                                        filter_parameters)
+        return {"total_count": search_count, "offset": offset, "limit": limit, "result": search_result}
+
+    def get_service_for_given_service_uuid(self):
+        service = ServiceRepository().get_service_for_given_service_uuid(self.org_uuid, self.service_uuid)
+        return service.to_dict()
+
+    def publish_service_data_to_ipfs(self):
+        service = ServiceRepository().get_service_for_given_service_uuid(self.org_uuid, self.service_uuid)
+        if service.service_state.state == ServiceStatus.APPROVED.value:
+            service_metadata = service.to_metadata()
+            filename = f"{service.uuid}_service_metadata.json"
+            service.metadata_ipfs_hash = ServicePublisherService.publish_to_ipfs(filename, service_metadata)
```

Comment:
publish to ipfs should stay in domain. i.e.: service.publish_to_ipfs(). same as it is writen for the organization.